src/Subscriber/EventSubscriber.php

```diff
@@ -33,46 +58,65 @@ public static function getSubscribedEvents(): array
 
     public function pushOrder(EntityWrittenEvent $event): void
     {
-        if (!isset($event->getIds()[0])) {
+        if (! isset($event->getIds()[0])) {
             return;
         }
 
-        $id = $event->getIds()[0];
-
+        $debug = false;
         try {
+            $order = $this->orderRepository->search(new Criteria([$event->getIds()[0]]), $event->getContext())->first();
+            if (! $order instanceof OrderEntity) {
+                return;
+            }
+
+            $salesChannelId = $order->getSalesChannelId();
+
+            $disabled = $this->configService->getBool($this->buildConfigKey('disabled'), $salesChannelId);
+            if ($disabled) {
+                return;
+            }
+
+            $subdomain = $this->configService->getString($this->buildConfigKey('subdomain'), $salesChannelId);
+            $connectionKey = $this->configService->getString($this->buildConfigKey('connectionkey'), $salesChannelId);
+            $debug = $this->configService->getBool($this->buildConfigKey('debug'), $salesChannelId);
+
+            if (empty($subdomain) || empty($connectionKey)) {
+                throw new IncompleteConfigurationException($subdomain, $connectionKey);
+            }
+
             $this->client->pushOrder(
-                $this->getConfigurationValue('subdomain'),
-                $this->getConfigurationValue('connectionkey'),
-                $id
+                $subdomain,
+                $connectionKey,
+                $order->getId()
             );
-        } catch (InvalidConfigException $e) {
-            /*
-            * It is possible that the configuration values are not yet set
-            * due to the fact that a plugin needs to be installed and activated before you can configure these.
-            *
-            * Picqer will check the webshop on interval as well.
-            * When a call fails there might be a delay in orders showing up in Picqer, but they will not get lost.
-            *
-            * We choose to silently fail to prevent any disturbances of the webshop.
-            */
-
-            $this->logger->info('[Picqer] Plugin configuration is invalid', [
+        } catch (IncompleteConfigurationException $e) {
+            if (! $debug) {
+                return;
+            }
+
+            $this->logger->error('[Picqer] Subdomain and/or connection-key not configured', [
+                'subdomain' => $e->getSubdomain(),
+                'connectionKey' => $e->getConnectionKey(),
+            ]);
+        } catch (RequestFailedException $e) {
+            if (! $debug) {
+                return;
+            }
+
+            $this->logger->error('[Picqer] Could not call webhook', [
+                'endpoint' => $e->getEndpoint(),
```

Comment:
Als er iets mis gaat dan is het denk ik het handig als je hier het order id en wellicht ook salesChannelId id aan toevoegt.